samcli/commands/build/build_context.py

```diff
@@ -64,10 +64,14 @@ def __init__(
         self._container_manager: Optional[ContainerManager] = None
 
     def __enter__(self) -> "BuildContext":
-        self._template_dict = get_template_data(self._template_file)
 
-        self._function_provider = SamFunctionProvider(self._template_dict, self._parameter_overrides)
-        self._layer_provider = SamLayerProvider(self._template_dict, self._parameter_overrides)
+        stacks = SamBuildableStackProvider.get_buildable_stacks(
+            self._template_file, parameter_overrides=self._parameter_overrides
+        )
+        self._template_dict = [stack for stack in stacks if stack.is_root_stack][0].template_dict
```

Comment:
I guess this code piece is finding the root stack and returning its template. But accessing first element directly could be problematic, I would say populate and array and then check of array has 1 element (I guess it should only have 1) and then access it. Raise a meaningful exception otherwise

Reply:
Done! thx